source/docs/software/wpilib-tools/wpical/index.rst

```diff
@@ -0,0 +1,192 @@
+
+There are a couple of common fields that both :guilabel:`OpenCV` and :guilabel:`MRcal` calibration options use:
+
+* :guilabel:`Select Camera Calibration Video` is the button to select your video file.
+
+* :guilabel:`Square Width` is the width of the squares (in inches) of the board.
+
+* :guilabel:`Marker Width` is the width of the ChArUco markers (in inches) on the board
+
+* :guilabel:`Board Width` is the width (in squares) of the board
+
+* :guilabel:`Board Height` is the height (in squares) of the board
+
+For :guilabel:`MRcal`, there are two more options to fill in:
+
+* :guilabel:`Image Width` is the width (in pixels) of the video's resolution
+* :guilabel:`Image Height` is the height (in pixels) of the video's resolution
+
+.. image:: images/MRcal.png
+    :alt: MRcal
+
+.. important:: For the calibration to work properly, it is important that you enter the exact width of the squares (and markers for ChArUco). It is recommended to use calipers to accurately measure these widths.
+
+When all the calibration fields have been entered, select :guilabel:`Calibrate` to run the calibration. A window will appear showing you the corner detection process:
+
+ChArUco:
+
+.. image:: images/ChArUcoDetection.png
+    :alt: ChArUcoDetection
+
+Checkerboard:
+
+.. image:: images/ChessboardDetection.png
+    :alt: CheckerboardDetection
+
+.. warning:: If any frames from the camera calibration look suspect, take a new video of the calibration board and try again.
+
+The camera calibration will automatically load the generated camera intrinsics JSON into WPIcal to continue to field calibration, but will also output the JSON file so it can be used for future calibrations. The calibration JSON will be saved as ``cameracalibration.json`` in the directory containing the calibration video you provided to WPIcal.
+
+External Camera Calibration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As opposed to calibrating cameras in the tool, WPIcal also allows you to upload a camera intrinsics JSON.
+
+CalibDB
+*******
+One common method for external camera calibration is `CalibDB.net <https://calibdb.net/>`_. WPIcal allows you to directly upload the generated CalibDB JSON without any modification. Make sure to download the OpenCV format JSON:
+
+.. image:: images/CalibdbDownload.png
+    :alt: CalibdbDownload
+
+Then:
+
+.. image:: images/CalibdbOpenCVFormat.png
+    :alt: CalibdbOpenCVFormat
+
+.. important:: When uploading a CalibDB JSON, ensure that you download the calibration with the proper resolution selected.
+
+.. warning:: CalibDB calibrations have been known to fail when bad snapshots are taken. If your calibration data looks suspect, repeat calibration process again.
+
+Custom JSON
```

Comment:
Do you think there's any value in supporting the photonvision JSON format directly? It's fairly similar

```
{
"resolution":{"width":1600.0,"height":1200.0},
"cameraIntrinsics":{"rows":3,"cols":3,"type":6,"data":[894.6473376999364,0.0,793.0568907406183,0.0,894.5611200668925,667.7549963404546,0.0,0.0,1.0]},
"distCoeffs":{"rows":1,"cols":8,"type":6,"data":[-0.04029854641739141,-0.03601719361588292,-0.0016269824645254384,-6.609780019207605E-4,0.14767766830679485,-0.09429815218469685,0.057468937992229234,0.11574241865507603]},
"calobjectWarp":[6.226950423907065E-4,6.020992541813042E-5]
}
```

Reply:
You mean if teams want to use their PV calibrations and cameras for field calibration?

Reply:
Yeah, but also i get why it would make sense to just keep wpical stuff within the wpical ecosystem. It'd be cool if everyone could just agree on a schema.